lib/decompress/zstd_decompress.rs

```diff
@@ -405,7 +403,10 @@ unsafe fn ZSTD_initLegacyStream(
             if dctx.is_null() {
                 return Error::memory_allocation.to_error_code();
             }
-            ZBUFFv05_decompressInitDictionary(dctx, dict.cast(), dictSize);
+            ZBUFFv05_decompressInitDictionary(
+                dctx,
+                core::slice::from_raw_parts(dict.cast(), dictSize),
```

Comment:
the `dict.is_null()` branch should have an `assert_eq!(dict_size, 0)` to make this safe (the assert is present in the original C code).